spec/ddtrace_integration_spec.rb

```diff
@@ -3,6 +3,13 @@
 
 RSpec.describe 'ddtrace integration' do
   context 'graceful shutdown', :integration do
+    before do
+      # TODO: Restore test for JRuby after further investigation on why it's flaky
+      # It currently breaks CI too frequently, with the only practical recourse
+      # being to rerun the build.
+      skip('Flaky test on JRuby. Requires further investigation.') if PlatformHelpers.jruby?
+    end
```

Comment:
I don't quite like calling JRuby out here -- the test is *flaky* on all Rubies (at least there's evidence to it), is just that well, like in many other Ruby threading bugs, JRuby with its parallel threads ends up uncovering those bugs more often than the other Rubies. (Thanks JRuby for keeping us honest!)

Aka I'd go with something like:

```suggestion
    before do
      # TODO: This test is flaky, and the flakiness affects JRuby really often.
      # Until we can investigate it, let's skip it as the constant failures impact unrelated development.
      skip('TODO: This test is flaky, and triggers very often on JRuby. Requires further investigation.') if PlatformHelpers.jruby?
    end
```